packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -121,8 +131,8 @@ export default class WebpackConfigGenerator {
     return defines;
   }
 
-  getMainConfig() {
-    const mainConfig = this.resolveConfig(this.pluginConfig.mainConfig);
+  async getMainConfig() {
```

Comment:
One of the main benefits of providing a function as webpack configuration is that you get `env` and `argv` passed in. Can we tie into them?

https://webpack.js.org/configuration/configuration-types/#exporting-a-function

Reply:
Additionally, it'd be useful to provide some arg (perhaps a custom env prop?) that describes if what we're building for -- `main`, `renderer`, or `preload`.